### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by Sumxx** [cocos/cocos-engine#5120]
https://github.com/cocos-creator/engine/blob/8474c229f5b8905c2935fc8a988770afb55f8fa2/cocos2d/core/renderer/render-engine.js#L13858-L13872

Why is there no equal sign? Is it intentional? No need to release the 0 serial number?

```
RenderData.free = function free (data) {
    if (data instanceof RenderData) {
      for (var i = data.length-1; i > 0; i--) {  // why Not = ?
        _dataPool.free(data._data[i]);
      }
      data._data.length = 0;
      data._indices.length = 0;
      data.material = null;
      data.uvDirty = true;
      data.vertDirty = true;
      data.vertexCount = 0;
      data.indiceCount = 0;
      _pool.free(data);
    }
  };
```

**Comment by CoderNeko** [comment]
It's very helpful to me！！！

**Comment by Kingwl** [comment]
It's very helpful to me too.

**Comment by DevilNeverCry** [comment]
It's very helpful to me too too.

**Comment by Sumxx** [comment]
oh my god    data.length ?  data._data.length ?

**Comment by SantyWang** [comment]
Sorry for the late reply, before you create this issue, this problem had been resolved in v2.1.2 . reference: https://github.com/cocos-creator/engine/pull/5025
